### DISCUSSION TITLE
Making an Automation which clicking a button emails a table

### DISCUSSION THREAD

**Discussion by swing18** [Budibase/budibase#6162] [Help] [answered]
I’m trying to implement an automation in my app that emails a table after clicking a submit button on top of it. 

To do this, I know I need to create an Automation where the trigger is app action, and the “Do this” is Send Email (SMTP). I’ve set up the necessary email parameters (except for the HTML Contents) but from here on out I know I’m missing some steps.

First, how do I link the button from my screen to this automation, and do I need to set any specific fields to do so? Or is it enough to go into my button, set “Define actions” to trigger automation, and then select my automation. 

Second, how do I ensure the table is included in the contents of my email? Do I need to add another part to the automation to query rows and then include the result in the HTML contents of my email? Or is there a direct line of HTML code I can put into the HTML Contents section and if so, what would it be.
Thank you!

**Comment by melohagan** [comment, marked as answer]
Hi @swing18 

You can pass bindings through to the automation: https://docs.budibase.com/docs/app-action

**Reply by swing18** [reply]
what would the handlebar code be under HTML contents to include the table?

**Reply by melohagan** [reply]
Hey @swing18 

If you have your button under a data provider, then you can pass through the array of table rows to the automation like so:
<img width="969" alt="Screenshot 2022-06-06 at 17 42 01" src="https://user-images.githubusercontent.com/101575380/172206012-aa9d0887-3a45-4cd0-a1a3-9b358932bc84.png">

I believe pagination will limit how much data can be sent through. 

Example JSON Array of a data table:
<img width="775" alt="Screenshot 2022-06-06 at 17 44 58" src="https://user-images.githubusercontent.com/101575380/172206554-3a15e73b-3e84-4773-981b-e412c90640e6.png">

Your automation:
<img width="471" alt="Screenshot 2022-06-06 at 17 45 32" src="https://user-images.githubusercontent.com/101575380/172206647-3fbdaf77-f87c-4d5f-9dc5-340f7445d2d0.png">
 
Then in the HTML contents you will need to use a JavaScript transformer to convert the JSON from the trigger field binding to a HTML table. 
You can try this code:
```
let tableRows = $("trigger.fields").table;
let html = "<table><tr>";
for (let row in tableRows[0]) {
	html += "<th>" + row + "</th>"
}
html += "</tr>"
for (let row of tableRows) {
	html += "<tr>"
	for (let value of Object.values(row)) {
		html += "<td>" + value + "</td>"
	}
	html += "</tr>"
}
html += "</table>"
return html;
```

**Reply by swing18** [reply]
Thanks @melohagan ! Using your code I've got the email working, but each charachter prints on a seperate line.  This Stack overflow post helped,(https://stackoverflow.com/questions/66646280/html-table-from-json-only-showing-1-character-per-line)so I changed the first line to: 

`let tableRows = JSON.parse($("trigger.fields").Table);`

and all the data appears in a big blob with no real formatting. Any suggestions on formatting it?

**Reply by melohagan** [reply]
Going by this StackOverflow post (https://stackoverflow.com/questions/4829254/best-practices-for-styling-html-emails) your best bet is to use inline styles. 
That'll require a bit of adjusting the code above
